module_utils/network_lsr/argument_validator.py

```diff
@@ -347,7 +347,7 @@ def _validate_impl(self, value, name):
                 raise ValidationError(name, "invalid key '%s'" % (setting))
             if isinstance(validator, ArgValidatorDeprecated):
                 setting = validator.deprecated_by
-                validator = self.nested.get(setting, None)
+                validator = self.nested[setting]
```

Comment:
I noticed that the line above it also raised a error,  should we also raise an error here ? 
```python
            try:
                validator = self.nested[setting]
            except KeyError:
                raise ValidationError(name, "invalid key '%s'" % (setting))
```

Reply:
no. If `self.nested[setting]` would access a non-existing key, then that is not a validation error. Instead, it would be a user error of how the `ArgValidator` instance was set up.

Certainly not a `ValidationError`, because it's not an error validating the input. Maybe a more suitable exception would be useful, but why? If this happens, it's a bug in the program. That's really bad already, and the only solution for that is fixing the bug, not raising a better exception. The `KeyError` we'd get here is find, it will lead to a crash, we'll see it, and will then fix the bug.